soroban-token-sdk/src/event.rs

```diff
@@ -0,0 +1,56 @@
+use soroban_sdk::{symbol_short, Address, Env, EnvBase, IntoVal, MuxedAddress, Symbol};
+
+pub struct Events {
+    env: Env,
+}
+
+impl Events {
+    #[inline(always)]
+    pub fn new(env: &Env) -> Events {
+        Events { env: env.clone() }
+    }
+
+    #[deprecated = "use soroban_sdk::token::Approve::publish"]
+    pub fn approve(&self, from: Address, to: Address, amount: i128, expiration_ledger: u32) {
```

Comment:
For each of the functions I have simply brought back the old functionality that was deleted in https://github.com/stellar/rs-soroban-sdk/pull/1489, and haven't rewritten the functions to use the new types. I think it'd be better to not introduce change here, and in the next major lets delete these functions.